server/routerlicious/packages/services/src/redis.ts

```diff
@@ -14,50 +14,64 @@ import { Lumberjack } from "@fluidframework/server-services-telemetry";
  */
 export class RedisCache implements ICache {
 	private readonly expireAfterSeconds: number = 60 * 60 * 24;
-	private readonly prefix: string = "page";
+	private readonly prefix: string;
+
 	constructor(private readonly client: Redis.default, parameters?: IRedisParameters) {
 		if (parameters?.expireAfterSeconds) {
 			this.expireAfterSeconds = parameters.expireAfterSeconds;
 		}
 
 		if (parameters?.prefix) {
 			this.prefix = parameters.prefix;
+		} else {
+			this.prefix = "";
+			winston.warn("A prefix for RedisCache was not included in the parameters.");
+			Lumberjack.warning("A prefix for RedisCache was not included in the parameters.");
 		}
 
 		client.on("error", (err) => {
 			winston.error("Error with Redis:", err);
 			Lumberjack.error("Error with Redis", undefined, err);
 		});
 	}
-	public async delete(key: string): Promise<boolean> {
-		try {
-			await this.client.del(this.getKey(key));
-			return true;
-		} catch (error) {
-			Lumberjack.error(`Error deleting from cache.`, undefined, error);
-			return false;
-		}
-	}
 
-	public async get(key: string): Promise<string> {
-		return this.client.get(this.getKey(key));
+	public async get<T>(key: string, prefixOverride?: string): Promise<T> {
+		const stringValue: string = await this.client.get(this.getKey(key, prefixOverride));
+		return JSON.parse(stringValue) as T;
 	}
 
-	public async set(key: string, value: string, expireAfterSeconds?: number): Promise<void> {
+	public async set<T>(
+		key: string,
+		value: T,
+		expireAfterSeconds: number = this.expireAfterSeconds,
+		prefixOverride?: string,
+	): Promise<void> {
 		const result = await this.client.set(
-			this.getKey(key),
-			value,
+			this.getKey(key, prefixOverride),
+			JSON.stringify(value),
 			"EX",
-			expireAfterSeconds ?? this.expireAfterSeconds,
+			expireAfterSeconds,
 		);
 		if (result !== "OK") {
 			throw new Error(result);
 		}
 	}
 
+	public async delete(key: string, prefixOverride?: string): Promise<boolean> {
+		try {
+			const keyToDelete: string = this.getKey(key, prefixOverride);
+			const result = await this.client.del(keyToDelete);
+			return result === 1;
+		} catch (error) {
+			Lumberjack.error(`Error deleting from cache.`, undefined, error);
+			return false;
+		}
+	}
+
 	public async incr(key: string): Promise<number> {
 		try {
```

Comment:
I'm just curious...why don't we accept `prefixOverride` for `incr` and `decr`?

Reply:
I initially didn't add to these, because I didn't see the functions being used anywhere (but kept them because other RedisCache implementations had them). Will add it though, for consistency :)